iaso/api/mobile/entity.py

```diff
@@ -117,20 +83,32 @@ class MobileEntityViewSet(ModelViewSet):
     remove_results_key_if_paginated = True
     filter_backends = [filters.OrderingFilter, DjangoFilterBackend, DeletionFilterBackend]
     pagination_class = LargeResultsSetPagination
+    permission_classes = [permissions.IsAuthenticated, HasPermission("menupermissions.iaso_entities")]
     lookup_field = "uuid"
 
     def get_serializer_class(self):
         return MobileEntitySerializer
 
     def get_queryset(self):
-        profile = self.request.user.iaso_profile
+        user = self.request.user
+        app_id = self.request.query_params.get("app_id")
+
+        queryset = Entity.objects.filter_for_user_and_app_id(user, app_id)
 
-        # This function alter the queryset by adding non_deleted_instances
-        queryset = filter_queryset_for_mobile_entity(Entity.objects.filter(account=profile.account), self.request)
+        if queryset:
+            try:
+                queryset = queryset.filter_for_mobile_entity(
+                    self.request.query_params.get("limit_date"), self.request.query_params.get("json_content")
+                )
+            except InvalidLimitDateError as e:
+                raise Http404(e)
+            except InvalidJsonContentError as e:
+                raise Http404(e)
```

Comment:
This should raise a 400 and not a 404. It's not that data could not be found, the query is just wrong